### PR TITLE
Add Hard sigmoid activation function

### PR DESCRIPTION
**Pull request by wingertge**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

Required to load the DbNet and PaddlePaddle OCR ONNX models used in RapidOCR.

### Changes

1. Add (configurable) hard sigmoid activation function
2. Add ONNX import support for the HardSigmoid node

### Testing

New unit tests and end-to-end ONNX tests.

### CODE REVIEW

File: crates/burn-import/onnx-tests/tests/hard_sigmoid/hard_sigmoid.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+
+# used to generate model: relu.onnx
```
Comment (antimora) [inline crates/burn-import/onnx-tests/tests/hard_sigmoid/hard_sigmoid.py:3]: relu.onnx -> hard_sigmoid.onnx

**Review by antimora — APPROVED** [review]
LGTM from my end!

One part that I am not sure about is if we need hard_sigmoid_backward function as well?

@nathanielsimard can you please confirm?

**Comment by wingertge** [comment]
> One part that I am not sure about is if we need hard_sigmoid_backward function as well?

The only other activation function with a scalar parameter is `leaky_relu` which also doesn't have a backwards function so I couldn't figure out how to implement it. If there is a way I'd be glad to add it.

**Review by laggui — COMMENTED** [review]
Thanks for adding the implementation all the way through ONNX import!

For the backward, I think it's fine for now. We can let autodiff handle it since it's implemented as a combination of other ops.

**Pull request merged**